### DISCUSSION TITLE
neighbors

### DISCUSSION THREAD

**Discussion by sdeion** [JoelForamitti/agentpy#46] [Q&A]
Hello! I am trying to build up ABM in the field of warehouse where AGVs, Automated Guided Vehicles, are running.

Please help me in appointing **only "right side" of the grid**.  Neighbors method, for example "neighbors = self.grid.neighbors(self, **distance=1**)" , appoints **all 8 neighbors surround the grid**. I would like to appoint **only one neghbor which is located at the right side of the grid.**

Thank you!

**Comment by jofmi** [comment]
Hi! You can use GridIter (https://agentpy.readthedocs.io/en/latest/reference_grid.html#agentpy.GridIter) to select agents from a specific area of the grid.
